### ISSUE TITLE
[SPIR-V] SPV_EXT_descriptor_heap: StructuredBuffer<T> from ResourceDescriptorHeap yields incorrect error: UAV support not implemented with non-emulated heaps

### ISSUE THREAD

**Issue by JackiBackiBoy** [microsoft/DirectXShaderCompiler#8806]
**Description**
When using SPV_EXT_descriptor_heap and SPV_KHR_untyped_pointers SPIR-V extensions attempting to acquire StructuredBuffer<T> from ResourceDescriptorHeap results in compile error: `UAV support not implemented with non-emulated heaps.`

**Steps to Reproduce**
Does not matter if ResourceDescriptorHeap index is non-uniform or constant:
```c++
StructuredBuffer<Foo> foo_buffer = ResourceDescriptorHeap[0]; // Results in compile error
```

Compilation flags:
```c++
const WCHAR* args[] = {
    L"-HV",
    L"2021",
    L"-I",
    <dir>,
    L"-E",
    <entry_point>,
    L"-T",
    L"vs_6_6",
    DXC_ARG_PACK_MATRIX_COLUMN_MAJOR,
    DXC_ARG_WARNINGS_ARE_ERRORS,
    DXC_ARG_ALL_RESOURCES_BOUND,
    L"-spirv",
    L"-fspv-target-env=vulkan1.3",
    L"-fspv-extension=SPV_EXT_descriptor_heap",
    L"-fspv-extension=SPV_KHR_untyped_pointers",
    L"-fspv-use-descriptor-heap",
    L"-fvk-use-scalar-layout",
    DXC_ARG_DEBUG,
    DXC_ARG_SKIP_OPTIMIZATIONS,
};
```

**Actual Behavior**
`error: UAV support not implemented with non-emulated heaps.`

**Environment**
- DXC version: 1.9(5402-0d3ee6b5)(1.9.0.5402) - 1.9.0.5402 (0d3ee6b55-dirty)
- Host Operating System: Windows 11 Pro 26200.9168

**Comment by Nielsbishere** [comment]
I encountered this too
